posts/2022-08-24-open-liberty-with-cargo-tracker.adoc

```diff
@@ -0,0 +1,131 @@
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ocseal
+author_github: https://github.com/ocseal
+seo-title: Try out Eclipse Cargo Tracker, an enterprise-level Jakarta EE application, with Open Liberty - OpenLiberty.io
+seo-description: Experience Open Liberty in an industrial environment by using a Liberty server to run Eclipse Cargo Tracker, an open-source application built to simulate real-world implementations of Jakarta EE.
+blog_description: "Experience Open Liberty in an industrial environment by using a Liberty server to run Eclipse Cargo Tracker, an open-source application built to simulate real-world implementations of Jakarta EE."
+open-graph-image: https://openliberty.io/img/cargo-tracker-home.png
+---
+= Try out Eclipse Cargo Tracker, an enterprise-level Jakarta EE application, with Open Liberty
+Chanun Ong <https://github.com/ocseal>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+As one of the top Java framework choices for enterprise-level applications, Jakarta EE is at the center of projects all over the globe. The https://eclipse-ee4j.github.io/cargotracker/[Eclipse Cargo Tracker] open-source application was designed to replicate a complex enterprise application by using core Jakarta EE features with professional development standards. Now, you can get a taste of Open Liberty in an industrial environment by running Cargo Tracker with an Open Liberty server, all on your own computer.
+
+== The Cargo Tracker application
+
+Cargo Tracker simulates an industrially-developed freight tracking web application built with Jakarta EE 8 and domain-driven design principles. The application uses key Jakarta EE features, including Faces, CDI, Enterprise Beans, Bean Validation, RESTful Web Services, and JSON Binding. It also uses Persistence, along with Batch and Messaging.
+
+Cargo Tracker books, tracks, updates, and routes cargo objects. Messaging events are logged to update cargo locations while persistence ensures that cargo status is not changed or removed when the application is shut down. Each portion of Cargo Tracker represents a step in a real-life shipment process: the customer tracks their cargo using the Public Tracking interface, the shipping company schedules and routes deliveries using the Administration interface, and port handlers log arriving and departing cargo using the Mobile Event Logger. 
+
+
+image::/img/blog/cargo-tracker-home.png[Cargo Tracker Home Page,width=70%,align="center"]
+
+
+Cargo Tracker was originally developed to work with the Payara runtime and had not been previously used with Open Liberty, but the application's functionality was perfectly preserved after the server switch. There were no issues across operating systems, as Ubuntu 18.04, Windows 10, and Mac OS Monterey 12.4 were all able to run Cargo Tracker identically. 
```

Comment:
is this interesting to readers? It's an important concern from the perspective of your project, but seems like an internal detail that wouldn't concern external users (unless there was an OS it _didn't_ work with). If CT origin with Payara is likely to be of interest to readers, or if we're obligated to to acknowledge it, fine. Otherwise I'd remove it as it seems like a digression from the main idea.

Reply:
Got it, that makes sense. Removed.